bundles/org.openhab.core/src/main/java/org/openhab/core/util/ColorUtil.java

```diff
@@ -285,7 +292,18 @@ public static HSBType xyToHsb(double[] xy, Gamut gamut) throws IllegalArgumentEx
     }
 
     /**
-     * Fine precision version of xyToHsb().
+     * Transform <a href="https://en.wikipedia.org/wiki/CIE_1931_color_space">CIE 1931</a> `xy` format to
+     * <a href="https://en.wikipedia.org/wiki/HSL_and_HSV">HSV</a> based {@link HSBType}, with the RGB elements NOT
+     * rounded in order to produce an HSB with finer precision..
+     *
+     * See <a href=
+     * "https://developers.meethue.com/develop/application-design-guidance/color-conversion-formulas-rgb-to-xy-and-back/">Hue
+     * developer portal</a>.
+     *
+     * @param xy the CIE 1931 xy color, x,y[,Y] between 0.0000 and 1.0000. <code>Y</code> value is optional.
+     * @param gamut the gamut supported by the light.
+     * @return the corresponding {@link HSBType} created from non rounded RGB values.
+     * @throws IllegalArgumentException when input array has wrong size or exceeds allowed value range.
      */
     public static HSBType xyToHsbFine(double[] xy, Gamut gamut) throws IllegalArgumentException {
```

Comment:
I don't think we should do that. Why don't we always use the exact calculation (i.e. the one you call `fine`?

Reply:
> Why don't we always use the exact calculation

Umm. I thought it was you insisting on non exact. Or??

Reply:
No, that must be a mis-understanding. I want a method that returns 0..255 as `int` because that is a very common use-case. And I don't want hundreds of methods that essentially return the same values, just with different types. For precise values we can use `PercentType` and whoever wants a double can call `rgb[0].doubleValue()`.

That aside: I don't see a real-world use-case for color values with a precision better than what we had before. openHAB is about home automation and I doubt that there is a single device out there that can reliably produce light with an RGB color of 255,127,98 in a way that a human eye can see a difference to 254,127,98. I also don't see why it is important that a conversion "xyY - something - xyY" results in the same values as long as the color that is associated with these values is so similar that a human eye can't detect a difference. I'm willing to accept such code only because it does not add additional processing.

Reply:
@J-N-K I am sorry but we seem to be dancing in circles. So let me summarise below the method signatures that you do, resp. do not want in ColorUtil. => Is this correct?

### Wanted methods
```
public static HSBType xyToHsb(double[] xy) // see question 1. below
public static HSBType xyToHsb(double[] xy, Gamut gamut) // see question 1. below

public static int[] hsbToRgb(HSBType hsb)
public static PercentType[] hsbToRgbPercent(HSBType hsb) // see question 2. below
public static int hsbTosRgb(HSBType hsb)

public static double[] hsbToXY(HSBType hsb)
public static double[] hsbToXY(HSBType hsb, Gamut gamut)

public static HSBType rgbToHsb(int[] rgb) // see questions 1. and 3. below
public static HSBType rgbToHsb(PercentType[] rgb) // see questions 1. and 3. below
```
### Unwanted methods
```
private static HSBType xyToHsbFine(double[] xy, Gamut gamut)
private static double[] xyToRgb(double[] xy, Gamut gamut)
```
### Qustions
And if the above is the case, then I have two further questions for clarification..
1. Do you accept that it is OK for `blahblahToHsb()` to produce an HSB with non integer internal parameters?
2. Should we rename `hsbToRgbPercent()` as `hsbToRgb()` to be internally self consistent?
3. Or, alternatively should we rename `rgbToHsb()` as `rgbToHsbPercent()` or `rgbPercentToHsb()` to be internally self consistent?

Reply:
1. Sure. Why should there be a limitation on the internal representation in `HSBType`?
2. That would be nice, but it's not possible, because we can't have to methods with the same name and parameters that return different types.
3. Like you, I'm not so happy with the naming of `hsbToRgbPercent` and `hsbTosRgb` but I think we have to live with that because of the limitations of Java. I would prefer to keep the shorter names, they are more intuitive.

Reply:
Ok.